compiler/src/dotty/tools/dotc/interactive/Completion.scala

```diff
@@ -386,6 +365,61 @@ object Completion {
         !name.isConstructorName && name.toTermName.info.kind == SimpleNameKind
       def isStable = true
     }
+
+    object CompletionScope {
+      def from(denotations: Seq[SingleDenotation])(using Context): CompletionScope = {
+        val mappings = denotations
+          .filter(den => include(den.symbol, den.name))
+          .toList
+          .groupBy(_.name)
+          .map((name, denots) => name.stripModuleClassSuffix -> denots)
+        CompletionScope(mappings)
+      }
+
+      def fromNamed(namedDenotations: Seq[(Name, SingleDenotation)])(using Context): CompletionScope = {
+        val mappings = namedDenotations
+          .filter((name, denot) => include(denot.symbol, name))
+          .toList
+          .groupBy(_._1)
+          .map((name, namedDenots) => name.stripModuleClassSuffix -> namedDenots.map(_._2))
+        CompletionScope(mappings)
+      }
+
+      val empty = CompletionScope()
+    }
+
+    /** A wrapper over a map from names to their denotations
+     *  taking care of filtering out not matching completions
```

Comment:
I cannot find corresponding logic about `filtering` in the class. It is not the responsibility of the class?